plugins/timeline/index.js

```diff
@@ -0,0 +1,321 @@
+      },
+      isEventEnd: function(event) {
+        return /INFO - Done:/.test(event);
+      },
+      extractCommand: function(event) {
+        // Messages are of the form
+        // timestamp INFO - Executing: [command: details; [params]]
+        return /Executing: \[([^:^\]]*)/.exec(event)[1];
+      },
+      extractTimestamp: function(event) {
+        // Timestamps begin the line and are formatted as
+        // HH:MM:SS.SSS
+        // We don't care about the date so just set it to 0.
+        return Date.parse('01 Jan 1970 ' + event.slice(0, 12)); 
+      }
+    };
+  } else {
+    // This is a ChromeDriver log.
+    actions = {
+      isEventStart: function(event) {
+        return /: COMMAND/.test(event);
+      },
+      isEventEnd: function(event) {
+        return /: RESPONSE/.test(event);
+      },
+      extractCommand: function(event) {
+        return /: COMMAND ([^\s]*)/.exec(event)[1];
+      },
+      extractTimestamp: function(event) {
+        return parseFloat(/^\[?([^\]]*)/.exec(event)[1]) * 1000;
+      }
+    };
+  }
+
+  return TimelinePlugin.parseLog(logLines, sourceName, actions, referenceStart);
+};
+
+TimelinePlugin.parseLog =
+    function(entries, sourceName, actions, referenceStart) {
+  var parsedTimeline = [];
+  var currentEvent = {};
+  var index = 0;
+  var relativeStartTime = 0;
+  for (var j = 0; j < entries.length; ++j) {
+    var event = entries[j];
+    if (actions.isEventStart(event)) {
+      currentEvent = {
+        source: sourceName,
+        id: index++,
+        command: actions.extractCommand(event),
+        start: actions.extractTimestamp(event)
+      };
+      if (!relativeStartTime &&
+          currentEvent.command.toString() == 'setScriptTimeout' ||
+          currentEvent.command.toString() == 'set script timeout' ||
+          // [sic], the timeoutt typo is present in the logs
+          currentEvent.command.toString() == 'set script timeoutt' ||
+          currentEvent.command.toString() == 'SetScriptTimeout') {
+        relativeStartTime = currentEvent.start;
+      }
+    } else if (actions.isEventEnd(event)) {
+      currentEvent.end = actions.extractTimestamp(event);
+      currentEvent.duration = currentEvent.end - currentEvent.start;
+      parsedTimeline.push(currentEvent);
+    }
+  }
+
+  // Make all the times relative to the first time log types is fetched.
+  for (var k = 0; k < parsedTimeline.length; ++k) {
+    parsedTimeline[k].start += (referenceStart - relativeStartTime);
+    parsedTimeline[k].end += (referenceStart - relativeStartTime);
+  }
+
+  return parsedTimeline;
+};
+
+TimelinePlugin.prototype.outputResults = function(done) {
+  try {
+    fs.mkdirSync(this.outdir);
+  } catch(e) {
+    if ( e.code != 'EEXIST' ) throw e;
+  }
+  var stream = fs.createReadStream(
+      path.join(__dirname, 'timeline/indextemplate.html'));
+  var outfile = path.join(this.outdir, 'timeline.json');
+  fs.writeFileSync(outfile, JSON.stringify(this.timeline));
```

Comment:
True, but it's not like there's any other work we're blocking, so I don't see a reason to make it async.